modules/data-entry/src/main/frontend/src/questionnaire/Form.jsx

```diff
@@ -169,13 +169,9 @@ function Form (props) {
         })
         setLastSaveStatus(undefined);
       } else {
-        // If the user is not logged in, offer to log in
-        const sessionInfo = window.Sling.getSessionInfo();
-        if (sessionInfo === null || sessionInfo.userID === 'anonymous') {
-          // On first attempt to save while logged out, set status to false to make button text inform user
-          setLastSaveStatus(false);
-
-        }
+        setErrorMessage("Saving the data failed");
+        openErrorDialog();
+        setLastSaveStatus(false);
       }
       })
```

Comment:
This fixes the problem without the need to "resolve" a failed request:

```suggestion
    }).catch((err) => {
      setErrorCode(0);
      setErrorMessage(err?.message);
      openErrorDialog();
      setLastSaveStatus(undefined);
    })
```